src/orchestrator.ts

```diff
@@ -78,7 +77,7 @@ export class Orchestrator {
         // Setup
         const gen = this.fn(context);
         const actions: IAction[][] = [];
-        let partialResult: Task | TaskSet;
+        let partialResult: Task | TaskSet | undefined;
```

Comment:
In what cases do we expect `partialResult` to be `undefined`? On line 86 it seems we assign it as either a `Task` or `TaskSet`. I apologize if this is a dumb question since I don't yet fully understand how the type system works in TypeScript.

Reply:
Good point. I might be able to restrict this further.

Reply:
It had to do with the reference to partialResult being used before being explicitly set (line 86). I have edited the code so this is no longer required.